statsmodels/base/model.py

```diff
@@ -474,6 +474,135 @@ def fit(self, start_params=None, method='newton', maxiter=100,
         return mlefit
 
 
+    def _fit_zeros(self, keep_index=None, start_params=None, return_auxiliary=False, **fit_kwds):
+        """experimental fit the model subject to zero constraints
+
+        Intended for internal use cases until we know what we need.
+        API will need to change to handle models with two exog.
+
+        This is essentially a simplified version of `fit_constrained`, and does not need
+        to use `offset`.
+
+        The estimation creates a new model with transformed design matrix,
+        exog, and converts the results back to the original parameterization.
+
+        Some subclasses could use a more efficient calculation than using a new model.
+
+        Parameters
+        ----------
+        keep_index : array_like (int or bool) or slice
```

Comment:
I don't think slice or bool will work, we are indexing in several places with `int`, and expanding with int for extra params